format/FormatNexusEigerDLS16MI03.py

```diff
@@ -0,0 +1,46 @@
+import calendar
+
+import h5py
+
+from dxtbx.format.FormatCBFMiniPilatusHelpers import get_pilatus_timestamp
+from dxtbx.format.FormatNexusEigerDLS16M import FormatNexusEigerDLS16M
+from dxtbx.format.nexus import h5str
+
+
+def get_start_time(filename):
+    with h5py.File(filename, mode="r") as f:
+        timestamp = h5str(f["/entry/start_time"][()]).rstrip("Z")
+    return get_pilatus_timestamp(timestamp)
+
+
+class FormatNexusEigerDLS16MI03(FormatNexusEigerDLS16M):
+    @staticmethod
+    def understand(image_file):
+        # Get the file handle
+        with h5py.File(image_file, "r") as handle:
+            name = h5str(FormatNexusEigerDLS16M.get_instrument_name(handle))
+            return name and name.upper() in {"DIAMOND BEAMLINE I03", "DLS I03"}
```

Comment:
Please could we add the datestamp gating in the `understand` method -> we don't need to even consider using this format method for pre-issue or post-issue data